great_expectations/data_context/data_context/context_factory.py

```diff
@@ -116,6 +117,14 @@ def get_evaluation_parameters_store(self) -> EvaluationParameterStore:
             )
         return self._project.evaluation_parameter_store
 
+    def get_datasources(self) -> DatasourceDict:
+        if not self._project:
+            raise RuntimeError(
+                "This action requires an active DataContext. "
+                + "Please call `get_context()` first, then try your action again."
+            )
+        return self._project.datasources
```

Comment:
We should combine this `datasources` attr with `sources` in some capacity. I need this dependency to grab the grandparent datasource when working with the validation config